src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -348,18 +348,73 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
+                    and response.features['SecurityType'] == 'TrustedLaunchSupported':
+                # will set default value of security_type
+                pass
+
         # image_reference is an ID now
+        image_reference = response.id
         image_reference = {'id': image_reference}
         if image_reference_lun is not None:
             image_reference['lun'] = image_reference_lun
 
     if gallery_image_reference is not None:
+        from azure.cli.command_modules.vm._vm_utils import is_shared_gallery_image_id, is_community_gallery_image_id
+        if is_shared_gallery_image_id(gallery_image_reference):
+            # the gallery image reference is shared gallery image id
+            from azure.cli.command_modules.vm._vm_utils import parse_shared_gallery_image_id
+            image_info = parse_shared_gallery_image_id(gallery_image_reference)
+            from ._client_factory import cf_shared_gallery_image
+            gallery_image_info = cf_shared_gallery_image(cmd.cli_ctx).get(
+                location=location, gallery_unique_name=image_info[0], gallery_image_name=image_info[1])
+        elif is_community_gallery_image_id(gallery_image_reference):
+            # the gallery image reference is community gallery image id
+            from azure.cli.command_modules.vm._vm_utils import parse_community_gallery_image_id
+            image_info = parse_community_gallery_image_id(gallery_image_reference)
+            from ._client_factory import cf_community_gallery_image
+            gallery_image_info = cf_community_gallery_image(cmd.cli_ctx).get(
+                location=location, public_gallery_name=image_info[0], gallery_image_name=image_info[1])
+        else:
+            # the gallery image reference is compute gallery image id
+            from msrestazure.tools import parse_resource_id
+            terms = parse_resource_id(gallery_image_reference)
+            gallery_image_reference_rg, gallery_name, gallery_image_definition = \
+                terms['resource_group'], terms['name'], terms['child_name_1']
+            client = _compute_client_factory(cmd.cli_ctx)
+            gallery_image_info = client.gallery_images.get(resource_group_name=gallery_image_reference_rg,
+                                                           gallery_name=gallery_name,
+                                                           gallery_image_name=gallery_image_definition)
+        if hasattr(gallery_image_info, 'hyper_v_generation') \
+                and gallery_image_info.hyper_v_generation == 'V2' and not gallery_image_reference_lun:
+            # will set default value of hyper_v_generation
+            logger.warning('Starting Build 2023 event, az disk create command '
+                           'will deploy Trusted Launch VM by default. To know more about Trusted Launch, '
+                           'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```

Comment:
> If for notification message, we can be prescriptive, with parameter names impacted, yes.

```suggestion
            logger.warning('Starting Build 2023 event, az disk create command '
                           'will deploy Trusted Launch VM by default. Please use `--security-type TrustedLaunch` in your scripts to avoid the breaking change. To know more about Trusted Launch, '
                           'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```